Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm

```diff
@@ -1016,6 +1016,44 @@ static void checkColor(WebCore::CocoaColor *color, std::optional<ColorExpectatio
     TestWebKitAPI::Util::run(&finished);
 }
 
+TEST(WritingTools, CompositionWithRestart)
+{
+    RetainPtr session = adoptNS([[WTSession alloc] initWithType:WTSessionTypeComposition textViewDelegate:nil]);
+
+    RetainPtr webView = adoptNS([[WritingToolsWKWebView alloc] initWithHTMLString:@"<body id='p' contenteditable><p id='first'>Hey do you wanna go see a movie this weekend - start</p></body>"]);
+    [webView focusDocumentBodyAndSelectAll];
+
+    __block bool finished = false;
+
+    [[webView writingToolsDelegate] willBeginWritingToolsSession:session.get() requestContexts:^(NSArray<WTContext *> *contexts) {
```

Comment:
super minor, but I've been starting to get into the pattern of putting as little stuff into the completion handlers as possible (see `IntelligenceTextEffectCoordinatorDelegate_TextPreviewsForRange`) since I think it makes it slightly more readable, but doesn't really matter

Reply:
Will keep this in mind moving forward, but I'm going to land this as-is for now.